vj4/handler/record.py

```diff
@@ -21,15 +22,13 @@
 from vj4.util import options
 
 
-class RecordMixin:
-  def tdoc_visible(self, tdoc):
-    now = datetime.datetime.utcnow()
-    if (not contest.RULES[tdoc['rule']].show_func(tdoc, now)
-        and (self.domain_id != tdoc['domain_id']
-             or not self.has_perm(builtin.PERM_VIEW_CONTEST_HIDDEN_SCOREBOARD))):
-      return False
-    return True
+class RecordVisibilityMixin(contestHandler.ContestVisibilityMixin):
+  async def rdoc_contest_visible(self, rdoc):
+    tdoc = await contest.get(rdoc['domain_id'], rdoc['tid'])
+    return self.can_show_record(tdoc), tdoc
 
+
+class RecordCommonOperationMixin(object):
```

Comment:
\- `(object)`

Reply:
this follows conventions elsewhere, for example, `class SettingMixin(object)` in `setting.py`.